server/src/main/java/io/druid/indexing/overlord/SegmentPublishResult.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.overlord;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.api.client.repackaged.com.google.common.base.Preconditions;
+import io.druid.timeline.DataSegment;
+
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Result of an operation that attempts to publish segments. Indicates the set of segments actually published
+ * and whether or not the transaction was a success.
+ *
+ * If "success" is false then the segments set will be empty.
```

Comment:
yes, since "segments" only includes segments actually published in this transaction- it's possible for the same segments to have been published in some other transaction (like if there was a replica)